src/System.Device.Gpio.Tests/GpioControllerTestBase.cs

```diff
@@ -169,17 +174,26 @@ public void ThrowsIfWritingClosedPin()
         }
 
         [Fact]
-        [Trait("SkipOnTestRun", "Windows_NT")] // Currently, the Windows Driver is defaulting to InputPullDown instead of Input when Closed/Opened.
-        public void OpenPinDefaultsModeToInput()
+        [Trait("SkipOnTestRun", "Windows_NT")] // Currently, the Windows Driver is defaulting to InputPullDown, and it seems this cannot be changed
+        public void OpenPinDefaultsModeToLastMode()
         {
-            using (GpioController controller = new GpioController(GetTestNumberingScheme(), GetTestDriver()))
+            var driver = GetTestDriver();
+            if (driver is SysFsDriver)
+            {
+                // See issue #1581. There seems to be a library-version issue or some other random cause for this test to act differently on different hardware.
+                return;
+            }
+
+            // This works for input/output on most systems, but not on pullup/down, since sometimes the hardware doesn't have read possibilities for those (ie. the Raspi 3)
+            using (GpioController controller = new GpioController(GetTestNumberingScheme(), driver))
             {
                 controller.OpenPin(OutputPin);
-                Assert.Equal(PinMode.Input, controller.GetPinMode(OutputPin));
                 controller.SetPinMode(OutputPin, PinMode.Output);
+                Assert.Equal(PinMode.Output, controller.GetPinMode(OutputPin));
```

Comment:
same comment about testing this for input and output

Reply:
in here might be a good idea to also write something to make sure it's not just a cached value of mode